app/models/concerns/submit_result.rb

```diff
@@ -0,0 +1,56 @@
+class SubmitResult
+  module SubmitResultType
+    SUCCESSFULL = "successfull"
+    HAS_SYNTAX_ERRORS = "has_syntax_errors"
+    PENDING_REFACTORING = "pending_refactoring"
+  end
+
+  attr_reader :syntax_errors
+  attr_reader :type
+
+  def initialize answer, keywords_hash
+    @syntax_errors = SubmitResult.findSyntaxErrors(answer)
+    @type = determineSubmitResult(answer, keywords_hash)
+  end
+
+  def is_successfull()
+    @type == SubmitResultType::SUCCESSFULL
```

Comment:
No `()` is necessary in the method name. Rails provides a nice `eql?` method for checking equality on strings